### ISSUE TITLE
Gradient Explosion Issue

### ISSUE THREAD

**Issue by luwis93choi** [luwis93choi/CNN_VO#2]
When training the network that is composed of only convolutional layer + normalization + activation + dropout, the network batch loss exploded on the 1st optimizer step.

**Comment by luwis93choi** [comment]
Applied the optimizer with L2 regularization in order to stem over-excited features.

When pooling layers are introduced in the network, the gradient explosion is diminished at learning rate 0.00001.

However, at higher learning rate, gradient explodes. It seems that for CNN, Visual Odometry problem is very sensitive to gradient shifts.

**Comment by luwis93choi** [comment]
Conclusion : This was due to inappropriate layer structure/composition of neural network.

Although visual odometry is very sensitive problem to gradient shifts (due to its sensitiveness of features), gradient explosion at learning rate 0.00001 is a highly abnormal phenomenon. This indicates past neural network structures was not up to the task of visual odometry.

After making CNN much deeper with 8 layers and 3 fully connected layers, gradient does not explode at learning rate 0.001.
